### PR TITLE
Refractor python version checking.

### PR DESCRIPTION
**Pull request by hmaarrfk**
~~Developers will be knowledgeable enough to know that it isn't python2
compatible.~~

~~The wheels won't be available in pypi or conda for python 2~~

This removes a bunch of visual noise from the `__init__.py` file.

## Checklist
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Unit tests

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by jni** [comment]
I'm -1 on this change. This will lead to a bunch of confusion for users. See [Python 3 Statement: Practicalities](https://python3statement.org/practicalities/) under "Fail early at import time". Feel free to move this into a utility function though, `assert_valid_python3_version()`. Actually we should check for Python >= 3.5, not just 3...

**Comment by hmaarrfk** [comment]
Good point.

I guess I stopped reading at "Fail early at install time".

I'm just really against the 20 lines of code this adds to what is supposed to be the starting point of the code. I know people don't like looking at source, but the source is really the only reliable form of information, and python makes it really to document the source.

Maybe a 
```
from .util.check_python_version import ensure_python3
```

**Comment by hmaarrfk** [comment]
@jni, see if you like this. If not, just close this PR.

You convinced me to keep this.

The other solutions would create more file pollution.

File: skimage/__init__.py [hunk 1]
```diff
@@ -68,14 +68,15 @@
 __version__ = '0.15dev'
 
 
-if sys.version_info < (3,):
+if sys.version_info < (3, 5):
```
Comment (soupault) [inline skimage/__init__.py]: JFYI, there was a discussion on the best approach for version comparison - https://github.com/scikit-image/scikit-image/pull/2975

Comment (jni) [reply skimage/__init__.py]: @soupault I think that discussion is more relevant to third-party packages where the version string is less predictable than Python system versions

**Comment by hmaarrfk** [comment]
@jni this is what the output looks like on a conda python2 environment:

```python
(py2) \git\scikit-image>python
Python 2.7.15 | packaged by conda-forge | (default, May  8 2018, 14:46:56) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import skimage
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "skimage\__init__.py", line 71, in <module>
    check_python_version((3, 5))
  File "skimage\_shared\version_requirements.py", line 28, in check_python_version
    """ % '.'.join([str(v) for v in min_version]))
ImportError:

You are running scikit-image on an unsupported version of Python.
Likely Python 2.

Unfortunately, scikit-image 0.15 and above no longer work on this
version of Python.  You therefore have two options: either upgrade to
Python 3.5, or install an older version of scikit-image. For Python 2.7, use

 $ pip install 'scikit-image<0.15'

Please also consider updating `pip` and `setuptools`:

 $ pip install pip setuptools --upgrade

Newer versions of these tools avoid installing packages incompatible
with your version of Python.

>>>
```

File: skimage/_shared/version_requirements.py [hunk 2]
```diff
@@ -4,6 +4,30 @@
 import sys
 
 
+def check_python_version(min_version):
+    if not isinstance(min_version, tuple):
```
Comment (hmaarrfk) [inline skimage/_shared/version_requirements.py]: I'm impartial to this, but it helps catch the case where you don't specify the python version as a tuple, but just as an int

**Comment by hmaarrfk** [comment]
Man, I didn't know that older pip didn't even check for version compatibility between python when pulling in sdist.

I had thought that this was caused by an adventurous user that just didn't listen to pip.

**Comment by jni** [comment]
> this is what the output looks like

Nice! The only change that I would suggest is not to guess their version ("Likely Python 2"), and instead spell it out: "You are using Python x.x, but the latest scikit-image requires version 3.5 or higher." Python 3.4 users are unlikely but plausible.

**Comment by hmaarrfk** [comment]
Yeah, I can do 

```
from platform import python_version
python_version()
```

should create a nicely formatted string.

The issue then is that the error message needs to branch:
  1. For 2.7/3/4, the last supported version is 0.14.x
  2. For other versions, please check our documentation on http://scikit-image.org/ to find the latest supported version.

**Comment by jni** [comment]
@hmaarrfk oh no, no branching, just ignore <= 2.6. We dropped support for that years ago, and haven't had any complaints.

**Comment by hmaarrfk** [comment]
New wording:

```
(py2) C:\Users\mark\git\scikit-image>python
Python 2.7.15 | packaged by conda-forge | (default, May  8 2018, 14:46:56) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import skimage
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "skimage\__init__.py", line 71, in <module>
    check_python_version((3, 5))
  File "skimage\_shared\version_requirements.py", line 30, in check_python_version
    """ % (python_version(), '.'.join([str(v) for v in min_version])))
ImportError:

You are running scikit-image on an unsupported version of Python.
Likely Python 2.

Unfortunately, scikit-image 0.15 and above no longer work with your installed
version of Python (2.7.15).  You therefore have two options: either upgrade to
Python 3.5, or install an older version of scikit-image.

For Python 2.7 or Python 3.4, use

 $ pip install 'scikit-image<0.15'

Please also consider updating `pip` and `setuptools`:

 $ pip install pip setuptools --upgrade

Newer versions of these tools avoid installing packages incompatible
with your version of Python.
```

**Comment by jni** [comment]
:+1: !

@scikit-image/core just need one more OK! ;)

File: skimage/_shared/version_requirements.py [hunk 3]
```diff
@@ -1,9 +1,36 @@
 from distutils.version import LooseVersion
+from platform import python_version
 import functools
 import re
 import sys
 
 
+def check_python_version(min_version):
```
Comment (soupault) [inline skimage/_shared/version_requirements.py]: How about `ensure_python_version`?

File: skimage/_shared/version_requirements.py [hunk 4]
```diff
@@ -1,9 +1,36 @@
...
+    if not isinstance(min_version, tuple):
+        min_version = (min_version, )
+    if sys.version_info < min_version:
+        raise ImportError("""
+
+You are running scikit-image on an unsupported version of Python.
+Likely Python 2.
```
Comment (soupault) [inline skimage/_shared/version_requirements.py]: `Likely Python 2.` can be safely removed now, I suppose.

**Pull request merged**